n_helpers.c

```diff
@@ -86,20 +86,154 @@ NOTE_C_STATIC int ytodays(int year);
+        return ERRSTR("size cannot be NULL", c_err);
+    }
+
+    // Issue a "card.binary" request.
+    J *rsp = NoteRequestResponse(NoteNewRequest("card.binary"));
+    if (!rsp) {
+        NOTE_C_LOG_ERROR("unable to issue binary request");
+        return ERRSTR("unable to issue binary request", c_err);
+    }
+
+    // Ensure the transaction doesn't return an error and confirm the binary
+    // feature is available.
+    if (NoteResponseError(rsp)) {
+        const char *err = JGetString(rsp, "err");
+        NOTE_C_LOG_ERROR(err);
+        JDelete(rsp);
+        NOTE_C_LOG_ERROR("unexpected error received during handshake");
+        return ERRSTR("unexpected error received during handshake", c_bad);
+    }
+
+    // Examine "cobs" from the response to evaluate the space required to hold
+    // the COBS-encoded data to be received from the Notecard.
+    long int cobs = JGetInt(rsp, "cobs");
+    JDelete(rsp);
+    if (!cobs) {
+        // If cobs is 0, the required buffer length is 0 because there's nothing
+        // to receive.
+        *len = 0;
+    } else {
+        // Otherwise, the required length is cobs + 1: the binary data plus
+        // 1 byte for the terminating newline.
+        *len = cobs + 1;
+    }
+
+    return NULL;
+}
+
+//**************************************************************************/
+/*!
+  @brief  Reset the Notecard's binary buffer
+
+  @returns  NULL on success, else an error string pointer.
+
+  @note  This operation is necessary to clear the Notecard's binary buffer after
+         a binary object is received from the Notecard, or if the Notecard's
+         binary buffer has been left in an unknown state due to an error arising
+         from a binary transfer to the Notecard.
+ */
+/**************************************************************************/
+const char * NoteBinaryDataReset(void)
+{
+    J *req = NoteNewRequest("card.binary");
+    if (req) {
+        JAddBoolToObject(req, "delete", true);
+
+        // Ensure the transaction doesn't return an error.
+        J *rsp = NoteRequestResponse(req);
+        if (NoteResponseError(rsp)) {
+            NOTE_C_LOG_ERROR(JGetString(rsp,"err"));
+            JDelete(rsp);
+            NOTE_C_LOG_ERROR("failed to reset binary buffer");
+            return ERRSTR("failed to reset binary buffer", c_err);
+        }
+    } else {
+        NOTE_C_LOG_ERROR("unable to allocate request");
+        return ERRSTR("unable to allocate request", c_mem);
+    }
+
+    return NULL;
+}
+
 //**************************************************************************/
 /*!
   @brief  Decode a binary payload received from the Notecard.
+
   @param  inBuf The binary payload.
   @param  inLen The length of the binary payload.
   @param  outBuf The buffer to write the decoded payload to. This can be the
-                 same address as inBuf, allowing for in-place decoding.
+                 same address as inBuf, allowing for in place decoding.
```

Comment:
Shouldn't this be in-place, as it was before?

Reply:
https://ell.stackexchange.com/a/236898

I went looking before, but now I found the definitive answer, and yes it should be hyphenated, because it is an adjective.

The following demonstrate the correct usage:

- "in-place decoding"
- "decoding in place"

This is crazy to me :exploding_head:!